lib/resolvers.js

```diff
@@ -0,0 +1,97 @@
+class Resolvers {
+  constructor(context) {
+    this.Kuzzle = context.accessors.sdk;
+
+    this._resolversMap = {
+      Query: {
+        allBooks: async () => {
+          const results = await this.Kuzzle.document.search(
+            'test-data',
+            'books'
+          );
+
+          if (!results.hits.length) {
+            throw new Error('No books were found!');
+          }
+          return results.hits.map((hit) => {
+            return { id: hit._id, ...hit._source };
+          });
+        },
+        book: async (root, { id }) => {
+          const result = await this.Kuzzle.document.get(
+            'test-data',
+            'books',
+            id
+          );
+
+          return { id: result._id, ...result._source };
+        },
+        author: async (root, { name }) => {
+          const authorResults = await this.Kuzzle.document.search(
+            'test-data',
+            'authors',
+            {
+              query: {
+                match: {
+                  name: name,
+                },
+              },
+            }
+          );
+          if (!authorResults.hits.length) {
+            throw new Error(`Couldn't find an author with name ${name}`);
+          }
+          const author = authorResults.hits[0];
+          const books = await this.Kuzzle.document.search(
+            'test-data',
+            'books',
+            {
+              query: {
+                match: {
+                  authorId: author._id,
+                },
+              },
+            }
+          );
+
+          return {
```

Comment:
BTW you don't trigger an error if no authors are found, which is fine to me.

Reply:
Yep some detect errors and some let the internal exceptions leak, these examples are like the ones in the graphql-apollo documentation, proves that exceptions inside **resolvers** are automatically catch'd by graphql and sent as a response to the query, correctly added to the errors array.

Reply:
I'd go for the "Let GraphQL catch and format the errors for ya" option